hgtv/views/video.py

```diff
@@ -300,68 +302,60 @@ def video_edit(channel, playlist, video):
     """
     Edit video
     """
-    if playlist != video.playlist:
-        # This video isn't in this playlist. Redirect to canonical URL
-        return redirect(video.url_for('edit'))
-
+    current_speakers = [speaker.userid for speaker in video.speakers]
     form = VideoEditForm(obj=video)
-    formvideo = VideoVideoForm(obj=video)
-    formslides = VideoSlidesForm(obj=video)
-    formsync = VideoSlidesSyncForm(obj=video)
-    form_id = request.form.get('form.id')
-    if request.method == "POST":
-        if form_id == u'video':  # check whether done button is clicked
-            if form.validate_on_submit():
-                form.populate_obj(video)
-                video.make_name()
-                db.session.commit()
-                flash(u"Edited video '%s'." % video.title, 'success')
-                return render_redirect(video.url_for(), code=303)
-        elif form_id == u'video_url':  # check video_url was updated
-            if formvideo.validate_on_submit():
-                formvideo.populate_obj(video)
-                try:
-                    process_video(video, new=False)
-                except (DataProcessingError, ValueError) as e:
-                    flash(e.message, category="error")
-                db.session.commit()
-                return render_redirect(video.url_for('edit'), code=303)
-        elif form_id == u'slide_url':  # check slides_url was updated
-            if formslides.validate_on_submit():
-                formslides.populate_obj(video)
-                try:
-                    process_slides(video)
-                    if video.video_slides_mapping:
-                        video.video_slides_mapping_json = make_presentz_json(video, json.loads(video.video_slides_mapping))
-                except (DataProcessingError, ValueError) as e:
-                    flash(e.message, category="error")
-                db.session.commit()
-                return render_redirect(video.url_for('edit'), code=303)
-        elif form_id == u'video_slides_sync':
-            if formsync.validate_on_submit():
-                formsync.populate_obj(video)
-                try:
-                    if video.video_slides_mapping:
-                        video.video_slides_mapping_json = make_presentz_json(video, json.loads(video.video_slides_mapping))
+    if request.method == 'GET':
+        html_form = render_form(form=form, title="Edit Video", submit=u"Save",
+            cancel_url=video.url_for(), ajax=False, with_chrome=False)
+        return {'video': dict(video.current_access()), 'form': html_form}
+    if form.validate():
+        form.populate_obj(video)
+        if not playlist.name:
```

Comment:
Surely you need `if not video.name`?

Reply:
Huh. apparently this has been broken for 9 months now.